src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+
+        public Node(T value, Node<T> prev) {
```

Comment:
```suggestion
        private Node(T value, Node<T> prev) {
```
will this constructor be available from outside if the Node class is private?

Reply:
+